### DISCUSSION TITLE
Delete folders of expired links

### DISCUSSION THREAD

**Discussion by PeopleInside** [PrivateBin/PrivateBin#1447] [Q&A] [answered]
### Describe the problem/question

When a created link expire the folder still left empty on the server, there is a way to delete also folders of expired links?

### Did you use the FAQ section?

- [X] Yes, I have read [the FAQ](https://github.com/PrivateBin/PrivateBin/wiki/FAQ) and I found no solution/answer there.

### What you did?

1. Create a link that expire in 1 day
2. When link expire the php page is deleted but folders for that link still exist in the data folder

### What happens

When link expire the php page is deleted but folders for that link still exist in the data folder

### What should happen

Folder of expired link also be removed

### Server OS

Ubuntu

### Webserver

Apache

### PrivateBin version

1.7.4

### Browser and version

Firefox

### Local operating system and version

Windows 11

### Issue reproducibility

No, I cannot reproduce it on https://privatebin.net as I don't have SFTP access to check folders inside data :)

**Comment by elrido** [comment]
The included administration script comes with an `--empty-dirs` option that can handle that. See this page for more details on what it can do:
https://github.com/PrivateBin/docker-nginx-fpm-alpine?tab=readme-ov-file#running-administrative-scripts

**Reply by PeopleInside** [reply]
Hi, thank you for the reply!
I don't use docker or I don't understand well, from the guide what I should do.

The guide say about migrate, i don't won't migrate but have a cron that every day clean empty folder.
While i think is good the php file of expire links are deleted I don't like see many empty folder remain on the server, they made also hard find folders where the php file still be active because active links.

**Reply by elrido** [reply, marked as answer]
The script is not dependent on docker and part of the default PrivateBin distribution, not just the container image. You would simply call it from the PrivateBin folder, i.e. `./bin/administration --empty-dirs`

It is not doing any sophisticated things either, and you could simply call the `find data -type d -empty -delete` (delete any empty directories in data) command directly from your cronjob:
https://github.com/PrivateBin/PrivateBin/blob/9eadf09ac36afc30e6dc82e7a0501c1cdda8299a/bin/administration#L86

**Reply by PeopleInside** [reply]
I think to understand now, thank you!
Just create a cron-job that run ./bin/administration --empty-dirs I will try to set this, thanks.

Done and it works, thanks! 🎊
